codestodians/food-basket-front-end
# Add login functionality

#### What does this PR do?
- Implements the Login Functionality
- Separates the Header into presentational and container components.

#### Description of Task to be completed?
- Add error link configuration to the `apollo-client` for proper dev environment debugging.
- Separate the Login into presentational and container components. 
- Separate the `Header` component into presentational and connect components.
- Update tests for the `Header` and the `Login` components

#### How should this be manually tested?
- check out into the `feature/add-login-functionality-153670674` branch.
- Install dependencies with `yarn`
- Start the app with `yarn start`
- Login with details of an already-signed-up user.
- Run tests with `yarn test`

#### Extra details
- The reason for the separation of the `Header` component into distinct parts in this story was to make an attempt to boost the test coverage by independently testing the presentational and container components. Said effort has not yielded much fruit yet, however, for the sake of the already pre-existing standard, it is still right for the presentational and container components to be kept apart and tested independently.

#### What are the relevant pivotal tracker stories?
- [#153670674](https://www.pivotaltracker.com/story/show/153670674)
